src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -38,6 +56,13 @@ protected virtual IEnumerable<PropertyInfo> GetRegisteredProperties()
                 .GetProperties(BindingFlags.Public | BindingFlags.Instance)
                 .Where(p => AutoRegisteringHelper.IsEnabledForRegister(p.PropertyType, true));
         }
+
+        protected virtual IEnumerable<MethodInfo> GetRegisteredMethods()
+        {
+            return typeof(TSourceType)
+                .GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
```

Comment:
Why DeclaredOnly?

Reply:
The graphql spec does not support inheritance and so there will be better schema match

Reply:
It is all about implementation details here, not about spec.